T_QUINQUECOSTATUS_GENOME = "external_data/GCA_024222315.1_ASM2422231v1_genomic.fna.gz"

QUALITY_CONTROL_OUTPUT = [
    "results/qc/fastqc/THYME_607L_hifi.html",
    "results/qc/longqc/THYME_607L_hifi"
]
include: "raw_reads_qc.smk"

DATA_EXPLORATION_OUTPUT = "results/aligned_stats/binned_coverage.txt"
include: "data_exploration.smk"

RAGTAG_OUTDIR = "results/ragtag_scaffold"
SCAFFOLD_OUTPUT = [f"{RAGTAG_OUTDIR}/scaffold_10k_q30_GCA_024222315_infergaps/ragtag.scaffold.{suff}" for suff in ["fasta", "agp", "stats"]]
include: "ragtag.smk"

ASSEMBLY_QC_OUTPUT = [
    f"results/assembly_quality/txome_busco/{suffix}" for suffix in ["hifiasm_10k_q30_GCA_024222315_default_all", "hifiasm_10k_q30"]
],
include: "assembly_qc.smk"

SELECTED_SAMPLES_EXPERIMENT = [
    "SAa046", "SAa062", "THa252", "THb134", "Thym607"
]
ILLUMINA_VALIDATION_OUTPUT = [
    expand("results/bwa_illumina/{sample}_{hyb}.bam.{type}",
        type = ["stats", "flagstats"], hyb = ["hyb1", "hyb2"],
        sample = SELECTED_SAMPLES_EXPERIMENT
        ),
    expand(
            "analysis/03-count_illumina_reads_assembly/counts/{sample}_{hyb}.csv",
            hyb = ["hyb1", "hyb2"], sample = SELECTED_SAMPLES_EXPERIMENT
        )
]
include: "illumina_validation.smk"

MAP_CANDIDATES = [
    "results/bwa/candidates_THYME_607L_hifi.sam",
    "results/bwa/candidates_hifiasm_10k_q30.sam",
    "results/bwa/candidates_hifiasm_10k_q30_ragtag_all.sam",
]
include: "candidates.smk"

rule all:
    input:
        QUALITY_CONTROL_OUTPUT,
        DATA_EXPLORATION_OUTPUT,
        SCAFFOLD_OUTPUT,
        ASSEMBLY_QC_OUTPUT,
        ILLUMINA_VALIDATION_OUTPUT,
        MAP_CANDIDATES