---
title: "Chromosome-level assembly of *Thymus vulgaris*"
date: today
author: Curro Campuzano Jiménez
bibliography: "bibliography.bib"
fig-dpi: 400
format:
  revealjs: 
    slide-number: true
    chalkboard: 
      buttons: false
    preview-links: auto
    logo: images/birc.png
    css: styles.css
    footer: Project in Bioinformatics | MSc. in Bioinformatics at Aarhus University
resources:
  - demo.pdf
---

```{r}
library(tidyverse)
source('../analysis/00-miscellaneous/format.R')

```

## Thyme's Ecology in Saint-Martin-de-Londres

![Spatial distribution of *Thymus vulgaris* phenolic (black) and non-phenolic (yellow) chemotypes. Obtained from @thompsonPlantTraitsEcological2020](images/Thymusvulgarischemotypes_St%20Martin-de-Londres.png){#fig-spatial fig-align="center" width="65%"}

::: notes

An ecotype is a population (or subspecies or race) that is adapted to local environmental conditions. The implication is that those individuals which were best adapted to the prevailing conditions left the most offspring.

Chemotypes of *T. vulgaris* reflects the ecological gradient. There are two ecotypes, each associated with a different type of monoterpenes.

Below 200 meters, there are occasional winter frosts and non-phenolic chemotypes.

Above 250 meters, there are phenolic chemotypes (summer drought)
:::

## More complete scaffolds, but how?

::: columns
::: {.column width="40%"}
<br/>

We have a *de novo* assembly from HiFi, but it is highly fragmented

1. Reads as input
2. Contigs as input
:::

::: {.column width="60%"}
![Recently published *T. quinquecostatus* genome assembly](images/tq_paper.png){#fig-tq alt="Recently published T. quinquecostatus genome assembly" fig-align="center" width="90%"}
:::
:::

## Main idea

<br/>

### Can a [chromosome-level]{style="color:#8762ccff;"} assembly be obtained from a [*T. vulgaris*]{style="color:#4aac8dff;"} *de novo* assembly generated using HiFi reads by incorporating information from a [closely related genome]{style="color:#64ac48ff;"}?

# Data exploration {background="#4a3050"}

## How *T. vulgaris* reads map to *T. quinquecostatus genome?*

-   Most windows of *T. quinquecostatus* ($\sim 75\%$) correspond to Zero-process (repetitive sequences?)

$$
Y = \begin{cases} 0, & \textrm{with probability } p \\ \textrm{Poisson}(\lambda), & \textrm{with probability } 1-p \end{cases}
$$

## Is a homology-based assembly scaffolding approach effective without Hi-C data?

![](images/homology_scaffolding.png){fig-align="center"}

# Methods {background="#4a3050"}

##

![](images/simplified_methods.png){#fig-methods fig-align="center" width="100%"}

## Reproducible research

-   Chained execution of many CLI
-   Isolated software environment (using [Conda](https://docs.conda.io/en/latest/) and [Singularity](https://sylabs.io/))

![](images/snakemake.png){fig-align="center" width="30%"}

## Contribution to Snakemake project

![](images/PR_ragtag.png){fig-align="center" width="30%"}

# Results {background="#4a3050"}

## Scaffolding process with RagTag (77% placed nucleotides)

![Synteny-like plot of CM044164.1 (only four *T. vulgaris* contigs)](images/synteny.png){#fig-scaffold fig-align="center" width="90%"}

## Genome assembly size

```{r}
#| label: fig-size
#| fig-cap: 
#|   - "Comparison of genome assembly size"

library(ggtext)

tribble(
  ~name, ~size,
  "Only 13 pseudo-chromosomes",   695.63,
  #"Cytometric estimation",   754.60,
  "Including unplaced contigs",   911.87
) |>
  ggplot() +
  geom_segment(
    aes(y=name, yend=name, x=0, xend=size),
    size=1, color="#8762ccff") +
  geom_point(aes(y=name, x=size), size = 3, colour = "#8762ccff")+
  geom_vline(xintercept = 754.60, linetype = "dashed", color = "#cd5136ff")+
  annotate(
    "text", x=780, y=2, label="Cytometric estimation", angle=90)+
  theme_minimal()+
  ylab('') + xlab('*T. vulgaris* genome size (Mb)')+
  theme(axis.title.x = element_markdown())

  
```

## Quality assessment: N50 (contiguity)

From 1.87 Mb (n=133) to **48.92 Mb (n=8)**

![For [historical context]{style="color:#808080;"}, N50 statistics of the published plant genomes. Obtained from @sunTwentyYearsPlant2022a](images/contiguity.jpg){#fig-n50 fig-align="center" width="90%"}

## Quality assessment: [BUSCO](https://busco.ezlab.org/)

BUSCO looks for the presence of nearly-universal orthologous genes

Completeness score: 96.4%1 (using Eudicots dataset)

::: {.callout-warning appearance="simple"}
## Disclaimer

Complete and duplicated genes is **high** (31.4%)

-   Recent genome duplication event?

-   Chimeric assembly of haplotypes?
:::

## Aligned short reads

::: columns
::: {.column width="50%"}
![](images/satureja_montana.jpg){fig-align="center" width="75%"}
:::

::: {.column width="50%"}
![](images/thymus_vulgaris.jpg){fig-align="center" width="75%"}
:::
:::

::: footer
[Obatined from Herbari Virtual del Mediterrani Occidental](http://herbarivirtual.uib.es/)
:::

## Validation using experimental data

```{r}
library(ggtext) 
df <- read_csv("../analysis/03-count_illumina_reads_assembly/samtools_stats.csv")|>
  mutate(Hyb = as.factor(Hyb))

df$specie <-  df$Sample |>
  startsWith('T') |>
  ifelse('*T. vulgaris*', '*S. montana*') 

df |>
  filter(Hyb != 2)|>
  ggplot(aes(
    x = `Properly paired`, y = `error rate`,
    color = Hyb, shape = specie)
    )+
  # annotate each point with the sample name
  geom_text(aes(label = Sample), size = 4, vjust = -1, hjust = 0.7)+
  geom_point(size = 4)+
  xlab("Percentage of properly paired mapped reads")+
  ylab("# mismatches / # mapped nucleotides")+
  scale_color_discrete(name = "Hybridization condition", drop = FALSE)+
  guides(shape=guide_legend(ncol=1, fill=guide_legend(ncol=1)))+
  theme(legend.position="bottom")+
  ylim(c(0, 0.06))+
  xlim(60, 100)+
  theme(legend.text = element_markdown())
```

# Future work {background="#4a3050"}

##  {#future_work_list data-menu-title="Future work ideas"}

-   Genome assembly quality evaluation: **repetitive** sequences

-   Validation with experimental data: **transcriptome**

-   Improve assembly: **alternative** reference-based scaffold algorithms

-   Phylogenetic analysis: thyme **ecology**

# Conclussions {background="#4a3050"}

## Take-home message

<br/>

### [*De novo* assembly of HiFi reads + homology-based scaffolding]{style="color:#cd5136ff;"} is an effective approach to obtain a [chromosome-level]{style="color:#8762ccff;"} assembly but a more comprehensive [**evaluation**]{style="color:#64ac48ff;"} and experimental [**validation**]{style="color:#64ac48ff;"}
 has to be done
 
# Thanks!

## References {visibility="uncounted"}

::: {#refs}
:::

# Extra slides {background="#4a3050" visibility="uncounted"}

## Impact of Long-read sequencing technologies and Hi-C {visibility="uncounted"}

```{r}
#| label: fig-assemblies
#| fig-cap: 
#|   - "Evolution of the number of published **Chromosome-level genome assemblies**. Data was retrieved from [NCBI](https://www.ncbi.nlm.nih.gov/genome/) on 6/15/2023."
eu <- read_csv('eukaryotes_chrm.csv') |>
  separate_wider_delim('Organism Groups', delim = ';', names = c('Kingdom', 'Group', 'Subgroups'))|>
  arrange(`Release Date`)

hifi_date <- as.Date("2019-12-08")
hic_data <- as.Date("2009-09-08")
eu |>
  filter(`Release Date` > as.Date("2000-01-01"))|>
  filter(!Group %in% c('Protists', 'Other'))|>
#  filter(Group %in% c("Animals", "Plants"))|>
  mutate(n = cumsum(row_number()), .by = Group)|>
  ggplot(aes(x = `Release Date`, y = n, color = Group))+
  geom_smooth()+
  geom_vline(xintercept = as.POSIXct(hifi_date), linetype = "dashed", color = "#cd5136ff")+
  annotate("text", x=as.POSIXct(hifi_date +11^2), y=10^6, label="Hi-Fi paper publication", angle=90)+
  # geom_vline(xintercept = as.POSIXct(hic_data), linetype = "dashed", color = "#cd5136ff")+
  # annotate("text", x=as.POSIXct(hic_data +11^2), y=10^6, label="Hi-C paper publication", angle=90)+
  theme_bw()+
  guides(color = guide_legend(override.aes = list(fill = NA)),
         linetype = guide_legend(override.aes = list(fill = NA))) +
  theme(legend.key = element_rect(fill = "white"))+
  ylab('# Published chromosome-level assemblies')

```

::: notes
First, we assembly reads into contigs. Then, we assemble contigs into scaffolds, by joining contigs in the correct order and direction and joining them with gaps. If a scaffold is big enough that it corresponds to a chromosome, we call them pseudo-chromosomes.

How plausible is to obtain a chromosome-level assembly?

Fragments, mainly because of repetitive sequences

We do not have Hi-C (High Performance Chromosome Capture)
:::

## Hi-C {visibility="uncounted"}

![Obtained from Wikipedia](https://en.wikipedia.org/wiki/Hi-C_(genomic_analysis_technique)#/media/File:HiCschematic.png)

## Quality control of raw reads {visibility="uncounted"}

::: columns
::: {.column width="40%"}
-   23.65 Gb of HiFi long reads

-   31.34x coverage of *T. vulgaris* genome

-   Quality control was successful according to [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) and [LongQC](https://github.com/yfukasawa/LongQC)
:::

::: {.column width="60%"}
![Per sequence GC content. Obtained with [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc)](images/gc_fasqc.png){#fig-gc fig-align="center" width="90%"}
:::
:::

<br/>

## *De novo* assembly {visibility="uncounted"}

-   [HiFiasm](https://github.com/chhylp123/hifiasm) assembled $\sim 1.7\cdot 10^6$ long reads into $\sim1.900$ contigs

-   $N50= 1.87$Mb

</br>

. . .

::: callout-important
Highly fragmented, but within the expectation for highly repetitive plant genomes
:::

## Bayesian inference using Julia (Turing library) {visibility="uncounted"}

``` {.julia filename="model.jl"}
ZeroInflatedPoisson(λ, p) = MixtureModel(
  [Dirac(0), Poisson(λ)], [p, 1 - p]
  )
  
@model function zip(Y)
    α = 1
    θ = mean(y)
    λ ~ Gamma(α, θ)
    pzero ~ Uniform(0, 1)
    Y ~ ZeroInflatedPoisson(λ, pzero)
end;
```

## Whole-genome alignment between *T. vulgaris* and *T. quinquecostatus* {visibility="uncounted"}

![Covered areas of T. quinquecostatus in the Whole-genome alignment](images/whole_genome_alignment.png){#fig-whole fig-align="center" width="90%"}

## Infer gap sizes {visibility="uncounted"}

RagTag infers the gap size between two adjacent sequences, $\textrm{seq1}$ and $\textrm{seq2}$, according to Equation

$$
\textrm{gapsize}  = \left(\textrm{aln2}_\textrm{rs} - \textrm{aln2}_\textrm{qs}\right) - \left(\textrm{aln1}_\textrm{re} - \textrm{aln1}_\textrm{qe} + \textrm{len}(\textrm{seq1}\right))
$$

![](images/infer_gapsize.png){fig-align="center" width="100%"}



## Modelling *T. vulgaris* long reads aligned to *T. quinquecostatus* {visibility="uncounted"}

1.  Align best 5% of reads to *T. quinquecostatus* assembly and model mapped reads using Bayesian inference
2.  Count mapped reads into 1000-length windows (reduce computational power)
3.  Fit Zero inflated Poisson

$$
Y = \begin{cases} 0, & \textrm{with probability } p \\ \textrm{Poisson}(\lambda), & \textrm{with probability } 1-p \end{cases}
$$

## Bayesian inference {visibility="uncounted"}

```{r}
#| label: fig-posterior
#| fig-cap: 
#|   - "Sampled posterior distribution of parameters for pseudo-chromosome CM044164.1 using MCMC"
df <- read_csv("../analysis/01-mapped_reads-model/chains/CM044164.1.csv") |>
  pivot_longer(cols = c("λ", "pzero"))|>
  mutate(type = "Posterior")
set.seed(100)
pzero_prior <- tibble(
  name = 'pzero', value = runif(300, 0, 1), type = 'Prior'
  )

lambda_prior <- tibble(
  name = 'λ', value = rgamma(300, 0.5453453453453454), type = 'Prior'
  )
library(latex2exp)

data_plot <- bind_rows(df, pzero_prior, lambda_prior) |>
  mutate(name = as.factor(name))
levels(data_plot$name) <- c(TeX("$p$"), TeX("$\\lambda$"))

data_plot|>
  ggplot(aes(x = value, fill = type))+
  geom_density(alpha = 0.7)+
  facet_wrap(
    ~name,ncol = 1, scales = "free",
    labeller=label_parsed
    ) +
  theme_light()+
  theme(legend.position="bottom")+
  xlab("Value")+ylab('Density')+
  guides(fill=guide_legend(title=""))
```

## Validation using experimental data {visibility="uncounted"}

Two hybridization conditions to enrich certain sequences: [**15%**]{style="color:#ce6db5ff;"} and [**5%**]{style="color:#59af69ff;"} of nucleotide divergence [@bataillonGenotypePhenotypeGenetic2022]

## {visibility="uncounted"}


![Aligned short Illumina reads to the *T. vulgaris* assembly](images/illumina.png){#fig-illumina alt="Aligned short Illumina reads to the T. vulgaris assembly" fig-align="center" width="100%"}
