The vegetation found in the Mediterranean region can thrive in two contrasting environments: areas with severe drought stress and mild winters, as well as colder regions, such as mountains, with extreme freezing temperature events in winter. These, even single events, can have a significant impact due to the daily temperature oscillations. This duality conditions the evolution of Mediterranean plant traits.~\cite{thompsonBibliography2020}\\

As a result of a temperature inversion, winter temperatures are several degrees lower in the St Martin-de-Londres basin (located 25 km north of Montpellier) compared to the surrounding hills. The distribution of different \textit{T. vulgaris} chemotypes reflects this drastic ecological gradient.~\cite{thompsonBibliography2020,bataillonGenotypePhenotypeGenetic2022} \\

Non-phenolic chemotypes (geraniol, $\alpha$-terpineol andthuyanol-4, linalool) dominate below 200 meters, where occasional winter freezing excludes phenolic chemotypes. Conversely, phenolic ecotypes (carvacrol and thymol) dominate above 250 meters, whereas summer drought excludes non-phenolic chemotypes.~\cite{thompsonBibliography2020} The ecological roles of monoterpenes are various, and the effects on the local soil environment and plant communities vary with the respective monoterpenes.~\cite{bataillonGenotypePhenotypeGenetic2022} \\

However, during the last few years, extreme freezing events have drastically reduced due to, hypothetically, the ongoing climate warming. This fact presumably caused the observed rapid increase in the frequency of the phenolic ecotype in the area, which has been under study over the last 50 years.~\cite{thompsonBibliography2020,bataillonGenotypePhenotypeGenetic2022} \\

Previous studies have demonstrated that chemotype variation is strongly heritable and most likely controlled by a few \textit{loci}. Thomas \etal identified a few loci as the potential genes involved.~\cite{bataillonGenotypePhenotypeGenetic2022}   Understanding the genetics of this polymorphism is critical in understanding its effects on community dynamics. It can also provide insight into fundamental questions in genetics: How does a species maintain multiple adaptive traits even in the presence of gene flow? How do plant populations respond to a decrease in strong selection pressure?~\cite{bataillonGenotypePhenotypeGenetic2022}\\

Resolving the genome structure of \textit{T. vulgaris} would be crucial to unlocking the genetics that determines its chemotypes and other traits.  Improvements in third-generation or long-read technologies make genome sequencing faster and more affordable, and it is the current method of choice to generate highly contiguous plant genome assemblies.~\cite{puckerPlantGenomeSequence2022} PacBio HiFi reads are long, 10â€“25 kb, and highly accurate, over 99.5\%.~\cite{honHighlyAccurateLongread2020} With HiFi reads, we can obtain more accurate and complete assemblies without a correction step, as we need with noisy long reads.~\cite{puckerPlantGenomeSequence2022}\\

Among the algorithms available for de novo assembly using long reads, Hifiasm, designed explicitly for HiFi long reads, stands out~\cite{chengHaplotyperesolvedNovoAssembly2021}. This algorithm is "Overlap-Layout-Consensus" based, as opposed to \textit{De Bruijn} graph based algorithms, such as Wtdbg2.~\cite{huangAlignGraph2SimilarGenomeassisted2021} However, current algorithms can struggle to differentiate between genome repeats that approach the length of long reads, generating hundreds and thousands of incomplete contigs.~\cite{huangAlignGraph2SimilarGenomeassisted2021} \\

More complete contigs, ideally assembled at the chromosomal level~\cite{AssemblyTerminologyGenome}, would benefit the study of \textit{T. vulgaris}. Among other things, we could determine linkage groups, estimate distances between loci, and avoid the complexity of working with a fragmented genome of thousands of contigs~\cite{tamazianChromosomerReferencebasedGenome2016}.\\

There are experimental and computational approaches in the literature to solve these problems. For example, Hi-C data can be used to create representations of complete chromosomes known as pseudochromosomes. These pseudochromosomes consist of ordered contigs linked by gaps (a sequence of ambiguous bases, Ns). The gaps provide only information on the distance between specific sequences without knowing the interleaved sequence.~\cite{puckerPlantGenomeSequence2022} Computational approaches benefit from non requiring physical chromosome maps. Different algorithms in the literature try to solve this problem by using a closely related genome assembly. The reference provides approximate information for the target genome assembly that we can integrate to improve the quality of our assembly.~\cite{huangAlignGraph2SimilarGenomeassisted2021} \\

Most of these approaches use a \textit{de novo} assembly as a starting point, except those so-called reference-guided assembly methods~\cite{ReferenceguidedAssemblyFour,lischerReferenceguidedNovoAssembly2017}. Roughly, some methods reassemble the contigs, sometimes referred to as similar genome-assisted reassembly methods~\cite{baoAlignGraphAlgorithmSecondary2014,huangAlignGraph2SimilarGenomeassisted2021,baoReMILOReferenceAssisted2018}, and others order and arrange contigs (from now on scaffold), sometimes called synteny-based or homology-based scaffolding methods.~\cite{kimReferenceassistedChromosomeAssembly2013,kolmogorovRagout2022,alongeAutomatedAssemblyScaffolding2022}\\

RagTag is a suite of tools for genome assembly scaffolding. It performs a whole-genome alignment to the reference assembly and scaffolds the draft assembly into longer sequences based on it without modifying it. It also joins the initial contigs with Ns. The resulting scaffolds are conceptual and practically equivalent to the pseudo-chromosomes obtained by Hi-C data; hence, we will interchange the two terms. \\