\section{Quality Control of raw reads}

Overall, our results indicate that the length of the \ac{HiFi} raw reads met our expectations, and the atypical distribution of GC content did not appear to pose any significant issues.We consider these sequences to be of high quality and suitable for \textit{de novo} assembly. 

\section{Modeling reads of T. vulgaris to T. quinquecostatus reference genome}


The high proportion of \textit{T. vulgaris} long reads successfully mapped to the \textit{T. quinquecostatus} genome indicates a significant similarity between the two genomes. This suggests that a Homology-based scaffolding approach could be applicable. However, we wanted to investigate further to determine any potential limitations of the technique.\\


The zero-inflated Poisson model that we fitted may have the following biological interpretation. We hypothesize that the \textit{T. quinquecostatus} genome consists of conserved regions with \textit{T. vulgaris} and non-conserved regions evolving randomly. The number of \textit{T. vulgaris} long reads that map to the \textit{T. quinquecostatus} genome can be modeled based on combining two processes: a Poisson process in the conserved regions, and the probability of mapping to a non-conserved region is always zero. By setting this probability to zero, we are assuming that the mutation rate between the two species is such that, without selective pressure, the DNA sequence has lost its original similarity. \\

To reduce the computational cost of the Bayesian inference process, we chose to compute the number of mapped reads per 1000-window along the genome. Instead of counting the number of mapped positions across each window, which would result in a less informative Normal distribution due to the Central Limit Theorem\footnote{\url{https://en.wikipedia.org/wiki/Central_limit_theorem}}, we counted the number of mapped reads.\\

If we accept the above interpretation,  our results are consistent but dependent on the chromosome results. The expected percentage of unconserved (or unmappable) regions oscillates between 74\% and 78\%. The expected number of mapped counts per 1000-window is slightly higher than two.\\

The percentage of non-conserved sequences seems to agree with the percentage of repeated sequences noted in \textit{T. quinquecostatus}, which is 70.61\%\cite{sunChromosomelevelAssemblyAnalysis2022}. However, it would be unlikely that this is the only cause to explain the size of the non-conserved regions. The annotation of the repeated sequences in the scaffolded reference genome is not trivial and is beyond this project's scope. \\

A more detailed characterization should be made before making any assertions. The size of the window, although in our experiences, does not influence it, and the number of sequences to be used are unresolved issues. In conclusion, our results suggest that even a tiny subset of the long reads are valid for this analysis.

\section{De novo assembly of T. vulgaris and homology-based assembly scaffolding}

We show in \autoref{fig:coverage_long_reads} the distribution of aligned contigs of the \textit{de novo} assembly into the reference assembly.  A large part, 47\% of the \textit{T. quinquecostatus} genome, is not covered in the whole-genome alignment. These uncovered areas are mainly located where we assume the centromeres are located. This value is in line with the estimates made by Bayesian inference.\\

That the percentage of the uncovered genome is lower than inferred may have several explanations. First, we are aligning the contigs made from the reads instead of the reads, which could, at least in theory,  increase the size of the total alignment compared to each of the "individual" alignments. It may also indicate that, as expected, a subset of the top 5\% of the sequences does not represent the total.  More experiments would have to be performed to explain this difference.\\

The draft assembly, obtained by de novo assembly of all the long reads, is highly fragmented, consisting of 1,884 contigs. As far as we know, there needs to be a review in the literature addressing how fragmented we can expect the different algorithms to assemble plant genomes. As a reference, it has been shown that for \textit{Vitis vinifera} long reads, current algorithms can assemble them into 591-1066 contigs. The estimated 1C value of \textit{V. vinifera} is 392Mb, half of the \textit{T. vulgaris} one. The number of contigs is within expectations.\\

After scaffolding the assembly, the number of contigs is 1,065. However, the different contigs placed into the 13 pseudo-chromosomes account for  77\% of the assembly size. The reference genome we use to reconstruct the genome consists of 92\% of sequences placed in the pseudochromosomes; thus, it would be unrealistic to obtain a value higher than this. Hereafter, we consider it an excellent value. \\

The high value of N50 and the correspondence with the estimated chromosome size supports the idea that we have achieved, to some extent, chromosome-level assembly. The high \ac{BUSCO} value supports that we have a complete assembly, although its value is not influenced by the degree of "fragmentation" that we try to decrease with the scaffolding process. A more in-depth characterization would have to be done to evaluate the accuracy and completeness.  \\

The main limitation of this scaffolded assembly is that, in most cases, we could not infer the gap size between the contigs that compose each pseudo-chromosome. Out of all the gaps in the assembly, only 8.3\% had a known, non-arbitrary length. While we could have relaxed the requirements to include inferred gaps as valid, doing so would have compromised some of the benefits of our current homology-based scaffolding method, including traceability, interpretability, and high confidence in the alignment.\\

While more complex algorithms, such as AlignGraph2 \cite{huangAlignGraph2SimilarGenomeassisted2021}, may have resulted in fewer unknown gaps, they would not have preserved the original sequence of the de novo assembly and may have produced less conservative results with less interpretability and traceability. Overall, our current method prioritizes transparency and obtaining conservative results.\\

\section{Mapping short Illumina reads to scaffolded assembly}

The short Illumina reads were sequenced after a hybridization protocol, in which baits (biotinylated oligonucleotide) hybridize to some areas of interest, then enrich those same regions. The \textit{T. vulgaris} sequences align more accurately and with fewer mismatches to the \textit{T. vulgaris} genome than the \textit{S. Montana} sequences. No substantial differences in the percentage of aligned sequences were observed between different \textit{T. vulgaris} individuals. However, as expected, the lowest number of mismatches occurred when using sequences obtained from the same individual to construct the reference genome, reflecting intraspecies genetic variation.\\

Using a more stringent hybridization condition, hybridization two slightly increased the percentage of mapped \textit{T. vulgaris} sequences and the number of mismatches. This may indicate that hybridization 2 enriches the number of reads corresponding to regions with homology to the \textit{T. vulgaris} region of interest but also increases mismatches due to interspecies genetic variation. However, a larger number of individuals would need to be included in the experiment to obtain statistically robust results, which is beyond this project's scope.