
In this work, we have evaluated the effectiveness of \ac{HiFi} sequencing for the \textit{de novo} assembly of plant genomes and demonstrated the utility of homology-based scaffolding to reconstruct a fragmented assembly into a chromosome-level one using computational methods. \\


Further evaluation of the accuracy and completeness of our scaffolded genome assembly is recommended for future work. Annotating sequence repeats throughout the genome\cite{flynnRepeatModeler2AutomatedGenomic2019} or identifying the exon-intron structure of well-characterized conserved protein families\cite{parraCEGMAPipelineAccurately2007} would shed light on the assembly's limitations. \\

More rigorous validation of our assembly would necessitate additional experimental data, particularly on calculating the coverage of short Illumina reads by aligning them to the reference genome. Additionally, assessing the impact of unknown-sized gaps and testing alternative reference-based algorithms would be worthwhile. Finally, leveraging this assembly for future population genomics studies could provide novel insights to complement existing knowledge.\\


Our data consistently suggest that the de novo assembly we have generated is high quality, albeit fragmented. Based on several metrics, including N50, whole-genome alignment, and cytometric estimates of the \textit{T. vulgaris} genome size, we have compelling evidence that our assembly represents a chromosome-level assembly to some degree. Moreover, our results and other experiences have also validated the experimental hybridization protocol used by Thomas \etal. \cite{bataillonGenotypePhenotypeGenetic2022}\\

In summary, our chromosome-level genome assembly, obtained through a reproducible preprocessing pipeline and a transparent computational approach, represents a valuable resource for future population genetics studies to advance the state-of-the-art in \textit{T. vulgaris} ecology.\\