
In this work, we have evaluated the effectiveness of \ac{HiFi} sequencing for the \textit{de novo} assembly of plant genomes and demonstrated the utility of homology-based scaffolding to reconstruct a fragmented assembly into a chromosome-level one using computational methods. Our results, together with other experiences, have also served to validate the experimental hybridization protocol used by Thomas \etal. \cite{bataillonGenotypePhenotypeGenetic2022}\\

While the assembly has limitations, future work could focus on assessing its accuracy and completeness, annotating repeats, calculating coverage in different individuals, and determining CEGMA (Core Eukaryotic Genes Mapping Approach) values. It would also be valuable to evaluate the impact of unknown-sized gaps and test other reference-based algorithms. \\

We have obtained a sufficiently good genome assembly for future projects through a reproducible preprocessing pipeline. Moreover, we have opted for a conservative and transparent methodology, which is particularly valuable given the absence of experimental data to validate our results, emphasizing the importance of our computational approach.\\