Mediterranean thyme (\textit{Thymus vulgaris}) is a subject of study of evolutionary ecology. In the South of France, the distribution of different chemical phenotypes (hereafter chemotypes) is consistent with differences in the abiotic environment. How are these chemotypes maintained within species in such a close geographic area, and which are the ecological genetics underlying this phenomenon are relevant questions in the field.~\cite{bataillon_genotype_2022}\\

As with other non-model organisms, the absence of a sequenced and accurately annotated reference genome limits the scope of the possible population genomic studies. However, the advances in long-read sequencing technologies make plant genome sequencing more affordable and accessible.~\cite{pucker_plant_2022} \\

The primary motivation of this project is to obtain a chromosome-level assembly of \textit{T. vulgaris}. To achieve this, we made a de novo assembly using High-Fidelity (HiFi) PacBio long reads. We subsequently arranged the contigs we obtained into pseudochromosomes using a homology-based scaffolding approach, using the closely-related \textit{T. quinquecostatus} genome as a reference. In this report, I will try to answer the questions:

\begin{itemize}
    \item Is it possible to achieve a chromosome-level resolution using a closely related specie as a reference?
    \item How much confidence can we place in such an assembly?
\end{itemize}

