The data analysis includes many heterogeneous steps using different bioinformatics programs and scripting languages. To ensure reproducibility and transparency, we used the Snakemake workflow management system~\cite{molderSustainableDataAnalysis2021}, and all the code is hosted in a public repository on GitHub\footnote{ \url{https://github.com/currocam/BiRC-Thyme}}.

\section{Quality Control of raw reads}


We used two automated quality control tools: FastQC~\cite{BabrahamBioinformaticsFastQC}, a well-known general program, and LongQC~\cite{fukasawaLongQCQualityControl2020}, a specific program for genomic datasets generated by third-generation sequencers. We also trimmed adapters from \ac{HiFi} reads using LongQC. ~\cite{fukasawaLongQCQualityControl2020}

\section{Modeling reads of T. vulgaris to T. quinquecostatus reference genome}

We retrieved the \textit{T. quinquecostatus} genome assembly published by Sun \etal from NCBI (Accession identifier: PRJNA690675).~\cite{sunChromosomelevelAssemblyAnalysis2022} We aligned a subset of the the \textit{T. vulgaris} long reads to the \textit{T. quinquecostatus} assembly using Minimap2~\cite{liMinimap2PairwiseAlignment2018} with default parameters for \ac{HiFi} sequences. To construct the subset of long reads, we filtered by quality, keeping the sequences corresponding to- the best 5\% (measured in nucleotides) using the specific program for this purpose Filtlong.~\cite{wickRrwickFiltlong2023}\\



We calculated the number of mapped reads in 1000-long windows along the genome, $Y = [ y_1, y_2, \dots, y_w]$. This number, $Y$, was modeled according to the model specified by \eqref{eq:model}. We estimated the model parameters $\lambda$ and $p$ using Bayesian inference, by computing the posterior distribution given the observed data using a Markov Chain Monte Carlo (\ac{MCMC}) sampler, for which we use the Julia library Turing.~\cite{DBLP:conf/aistats/GeXG18} We used a relatively small number of iterations, 200 and 500, in the adaptive and sampling phase, respectively, and a single chain. \\

\begin{subequations}
\label{eq:model}
\begin{align}
Y \sim \textrm{ZIPoisson}(p, \lambda) \label{eq:model1}\\
p \sim \textrm{Uniform}(0, 1) \label{eq:model2}\\
\lambda \sim \textrm{Gamma}(1, \alpha)  \label{eq:model3}\\
\alpha = \textrm{average}(Y) \label{eq:model4}
\end{align}   
\end{subequations}


\section{De novo assembly of T. vulgaris}\label{sec:denovo}

We constructed a \textit{de novo} assembly from the \textit{T. vulgaris} long reads using Hifiasm~\cite{chengHaplotyperesolvedNovoAssembly2021} with default parameters. I am circumventing the process as I did not execute it personally.
        
\section{Homology-based assembly scaffolding}\label{sec:scaffold}

We scaffolded the \textit{de novo} assembly using the bioinformatic program RagTag \cite{alongeAutomatedAssemblyScaffolding2022}. In general terms, the process consisted of 
\begin{enumerate}
    \item Perform a whole-genome alignment of \textit{T. vulgaris} to \textit{T. quinquecostatus} assembly using Minimap2. 
    \item Order and orient the contigs into longer sequences (pseudo-chromosomes) according to the alignment. 
    \item Join adjacent contigs of \textit{T. vulgaris} with gaps of inferred size (see \autoref{sec:infergapsize}). 
\end{enumerate}

\section{Assembly quality assessment}

We collected quality metrics of both assemblies using \ac{BUSCO}~\cite{manniBUSCOAssessingGenomic2021}, based on the gene content of near-universal single-copy orthologs and assembly-stats~\cite{Assemblystats2023}, for technical metrics. We compared the size of the genome with cytometric estimations\cite{PlantDNACvalues}. 

\section{Mapping short Illumina reads to scaffolded assembly}\label{sec:illumina}

We aligned the reads of three specimens of \textit{T. vulgaris} and two of \textit{Satureja montana} sequenced with Illumina (see \autoref{tab:illumina_samples}). Two hybridization conditions were employed before the construction of sequencing libraries.\\

\begin{enumerate}
    \item Hybridization 1 is a standard capture protocol that, according to the manufacturer, supports up to 15\% nucleotide divergence between the bait's nucleotide sequence and the captured genomic DNA. 
    \item Hybridization 2 is a more stringent procedure developed for a more specific capture. It involves capture, washing of the captured library, and recapture (details for the specific steps are in the supplementary information and protocols of Bataillon \etal \cite{bataillonGenotypePhenotypeGenetic2022}). The expected nucleotide divergence supported between the bait and the genomic target is much lower and roughly estimated to be within 5\% of nucleotide divergence to \textit{Thymus vulgaris} (used for the original baits design).
\end{enumerate} 

\begin{table}[h!]
    \begin{minipage}{\linewidth}
    \renewcommand\thefootnote{\thempfootnote}
    \centering
    \begin{tabular}{@{}llll@{}}
    \toprule
    Sample Id & Scientific name                      & Year & Location     \\ \midrule
    THc607\footnote{Same invididual as the genome assembly}    & \textit{T. vulgaris}                          & 2022   &  Cefe Montpellier  \\
    THb134    & \textit{T. vulgaris}                 & 2022  & Mont St. Baudille   \\
    THa252    & \textit{T. vulgaris}                 & 2016-2017 & \\
    SAa062    & \textit{S. montana} & 2022 & Cabanne     \\
    SAa046    & \textit{S. montana} & 2022  & Roc Blanc    \\ \bottomrule
    \end{tabular}
    \caption{List of specimens sequenced with Illumina and mapped to assembly. }
    \label{tab:illumina_samples}
    \end{minipage}
    \end{table}

Since I received the demultiplexed Illumina reads, I will not dive into this process. For the alignment, we used BWA-MEM with default parameters. ~\cite{liAligningSequenceReads2013}

