The data analysis includes many heterogeneous steps using different bioinformatics programs and scripting languages. To ensure reproducibility and transparency, we used the Snakemake workflow management system~\cite{molderSustainableDataAnalysis2021}, and all the code is hosted in a public repository on GitHub (\url{https://github.com/currocam/BiRC-Thyme}).

\section*{Quality Control of raw reads}


We used two automated quality control tools: FastQC~\cite{BabrahamBioinformaticsFastQC}, a well-known general program, and LongQC~\cite{fukasawaLongQCQualityControl2020}, a specific program for genomic datasets generated by third-generation sequencers. We also trimmed adapters from \ac{HiFi} reads using LongQC. ~\cite{fukasawaLongQCQualityControl2020}

\section*{Exploratory analysis of long reads and reference genome}

As part of our exploratory analysis, we aimed to assess how well the long reads of \textit{T. vulgaris} align to the reference genome of \textit{T. quinquecostatus}. \\

We retrieved the \textit{T. quinquecostatus} genome assembly published by Sun \etal from NCBI (Accession identifier: PRJNA690675).~\cite{sunChromosomelevelAssemblyAnalysis2022} We aligned a subset of the \textit{T. vulgaris} long reads to the \textit{T. quinquecostatus} assembly using Minimap2~\cite{liMinimap2PairwiseAlignment2018} with default parameters for \ac{HiFi} sequences. To construct the subset of long reads, we filtered by quality, keeping the sequences corresponding to the best 5\% (measured in nucleotides) using the specific program for this purpose Filtlong.~\cite{wickRrwickFiltlong2023}\\

We measured the percentage of mapped reads, the distribution of alignment lengths, and the coverage using samtools.~\cite{danecekTwelveYearsSAMtools2021} We studied the distribution of the mapped reads across the \textit{T. quinquecostatus} genome. We computed the number of mapped reads in 1000-long windows, $Y = [ y_1, y_2, \dots, y_w]$ to accomplish this. We propose that mapped reads can be accurately modeled as a Zero-inflated Poisson process, as shown in Equation \eqref{eq:zip}. \\

\begin{equation}
\label{eq:zip}
Y = \begin{cases} 0, & \textrm{with probability } p \\ \textrm{Poisson}(\lambda), & \textrm{with probability } 1-p \end{cases}
\end{equation}

We estimated the missing parameters of the model, $\lambda$ and $p$, through Bayesian inference. We examined the posterior distribution of $\lambda$ and $p$ to evaluate the model's goodness. In the discussion section, we provide a potential explanation of the implications of the Zero-inflated Poisson model.\\

We computed the posterior distributions based on very uninformative priors and a \ac{MCMC} sampler with a single chain and a small number of iterations: 200 in the adaptive phase and 500 in the sampling phase. The entire model specifications are shown in Equation \eqref{eq:model}. We implemented the model using the probabilistic programming Turing library. ~\cite{DBLP:conf/aistats/GeXG18}

\begin{subequations}
\label{eq:model}
\begin{align}
Y \sim \textrm{ZIPoisson}(p, \lambda) \label{eq:model1}\\
p \sim \textrm{Uniform}(0, 1) \label{eq:model2}\\
\lambda \sim \textrm{Gamma}(1, \alpha)  \label{eq:model3}\\
\alpha = \textrm{average}(Y) \label{eq:model4}
\end{align}   
\end{subequations}

\section*{De novo assembly of T. vulgaris}\label{sec:denovo}

A \textit{de novo} assembly of \textit{T. vulgaris} was available. This assembly was made from the long reads using Hifiasm~\cite{chengHaplotyperesolvedNovoAssembly2021} with default parameters.

\section*{Homology-based assembly scaffolding}\label{sec:scaffold}

Using the- bioinformatic program RagTag\cite{alongeAutomatedAssemblyScaffolding2022}, we scaffolded the \textit{de novo} assembly. In general terms, the process consisted of 
\begin{enumerate}
    \item Perform a whole-genome alignment of \textit{T. vulgaris} against \textit{T. quinquecostatus} assembly using Minimap2. 
    \item Order and orient the contigs into longer sequences (scaffolds) according to the alignment. 
    \item Join adjacent contigs of \textit{T. vulgaris} with gaps of inferred size. 
\end{enumerate}

As established by Alonge \etal ~ \cite{alongeAutomatedAssemblyScaffolding2022}, RagTag infers the gap size between two adjacent sequences, $\textrm{seq1}$ and \textrm{seq2}, according to (\ref{eq:infergapsize})

\begin{equation}\label{eq:infergapsize}
\textrm{gapsize}  = \left(\textrm{aln2}_\textrm{rs} - \textrm{aln2}_\textrm{qs}\right) - \left(\textrm{aln1}_\textrm{re} - \textrm{aln1}_\textrm{qe} + \textrm{len}(\textrm{rs}\right)
\end{equation}

where $\textrm{rs}$, $\textrm{re}$, $\textrm{qs}$ and $\textrm{qe}$ denote the start and end position of the alignment in the reference and query sequence, respectively. If the gap size is smaller than 100 bp or bigger than 100,000 bp, the inserted gap is forced to be 100 bp, indicating an unknown gap according to the AGP specification.\cite{AGPSpecificationV2} 

\section*{Assembly quality assessment}

We used assembly-stats~\cite{Assemblystats2023} for collecting technical metrics, and we compared the size of the genome with cytometric estimations\cite{PlantDNACvalues}. We collected quality metrics of both assemblies using BUSCO~\cite{manniBUSCOAssessingGenomic2021}, based on the gene content of near-universal single-copy orthologs. We used the datasets eudicots\_odb10 (2023-04-05) and eukaryota\_odb10 (2023-04-05). However, we report only the former's results, corresponding to a clade \textit{T. vulgaris} belong. \\



\section*{Mapping short Illumina reads to scaffolded assembly}\label{sec:illumina}

The demultiplexed Illumina reads of three specimens of \textit{T. vulgaris} and two of \textit{Satureja montana} were available (see \autoref{tab:illumina_samples}), including reads from the same individual sequenced using \ac{HiFi}.\\

\begin{table}[h]
    \begin{minipage}{\linewidth}
    \renewcommand\thefootnote{\thempfootnote}
    \centering
    \begin{tabular}{@{}llll@{}}
    \toprule
    Sample Id & Scientific name & Location     \\ \midrule
    THc607\footnote{Same individual as the genome assembly}    & \textit{T. vulgaris}   &  Cefe Montpellier  \\
    THb134    & \textit{T. vulgaris} & Mont St. Baudille   \\
    THa252    & \textit{T. vulgaris}& \\
    SAa062    & \textit{S. montana} & Cabanne     \\
    SAa046    & \textit{S. montana}  & Roc Blanc    \\ \bottomrule
    \end{tabular}
    \caption{List of specimens sequenced with Illumina and mapped to assembly}
    \label{tab:illumina_samples}
    \end{minipage}
    \end{table}

Two hybridization conditions were employed before the construction of sequencing libraries.\\
\begin{enumerate}
    \item Hybridization 1 is a standard capture protocol that, according to the manufacturer, supports up to 15\% nucleotide divergence between the bait's (biotinylated oligonucleotide probes) nucleotide sequence and the captured genomic DNA. 
    \item Hybridization 2 is a more stringent procedure developed for a more specific capture (details for the specific steps are in the supplementary information and protocols of Bataillon \etal \cite{bataillonGenotypePhenotypeGenetic2022}). The expected nucleotide divergence supported between the bait and the genomic target is much lower and roughly estimated to be within five percent of nucleotide divergence to \textit{T. vulgaris} (used for the original bait design).
\end{enumerate} 

We aligned these reads to the scaffolded assembly using BWA-MEM with default parameters. ~\cite{liAligningSequenceReads2013}