\section{Quality Control of raw reads}

We used LongQC and FASTQC to calculate basic technical statistics and evaluate the quality of HiFi raw reads~\cite{fukasawaLongQCQualityControl2020,BabrahamBioinformaticsFastQC}. The length of the 1,747,253 raw reads follows a unimodal distribution with an average size of 13.5 kb. The longest read is 46.7 kb long. \\

The quality reports show no issues except for the GC distribution, which exhibited a bimodal distribution. Instead of the expected normal distribution, we observed two peaks, with the first peak around 45\%, corresponding to the GC distribution of the same individual sequenced via Illumina, and the second peak around 37\%. We trimmed 25 sequences at 3' and 5' using LongQC. Despite the minimal impact of these, we used the trimmed sequences for the remainder of the analysis. \\

\section{Modeling reads of T. vulgaris to T. quinquecostatus reference genome}

Although not included in this report, we experimented with different subset sizes, alignment tools, and window sizes. These experiments produced consistent results. Regarding the number of iterations and chains, various experiments with both experimental and simulated data justified a conservative choice, as we obtained equivalent results in all cases using fewer computational resources. The posterior sampling distribution obtained using Bayesian inference for each chromosome independently is shown in \autoref{fig:bayesian_posterior}. Notice that we are using the number of mapped reads per window and not the number of mapped nucleotides. We made this decision made to reduce the computational cost. \\

\graphicspath{{gfx/}}
\begin{sidewaysfigure}
\centering
\input{gfx/05-posterior_var_edited.pdf_tex}
\caption{The posterior sampling distributions obtained from modelling the number of mapped reads per 1000-windows according to the model specified by \eqref{eq:model} using MCMC. }    
\label{fig:bayesian_posterior}

\end{sidewaysfigure}    

\section{De novo assembly of T. vulgaris and homology-based assembly scaffolding}

As discussed in \autoref{sec:denovo} and \autoref{sec:scaffold}, we obtained a \textit{de novo} assembly from our HiFi long reads, hereafter draft assembly, and a scaffolded assembly using RagTag (see \autoref{fig:coverage}).The scaffolding process placed 880 contigs (77\% of base pairs) into bigger supercontigs.  Interestingly, a few of the contigs of the draft assembly aligned not to the \textit{T. quinquecostatus} pseudo-chromosomes but to unplaced contigs of it. \\

We generated 13 supercontigs by joining the adjacent contigs of the draft assembly that aligned to each of the 13 pseudo-chromosomes of \textit{T. quinquecostatus} (supported by Hi-C experimental data). Therefore,  these supercontigs are a reconstruction of the \textit{T. vulgaris} pseudo-chromosomes based on the pseudochromosome similarity in both \textit{T. quinquecostatus} and \textit{T. vulgaris}. We show a selection of global statistics comparing the draft and the scaffolded assembly in \autoref{tab:global}. \\

\begin{table}[h!]
    \begin{minipage}{\linewidth}
    \renewcommand\thefootnote{\thempfootnote}
    \centering
    \begin{tabular}{@{}cccccc@{}}
        \toprule
        Metric              & N\footnote{Number of scaffolds}    & Ave (Mb)\footnote{Average length of the scaffolds} & Largest (Mb)\footnote{Largest scaffold} & N50 (Mb)\footnote{Length such that scaffolds of this length or longer include half the bases of the assembly. The number of scaffold, $n$ is also included.}     & Ns\footnote{Number of Ns ambiguous nucleotides}      \\ \midrule
        Draft assembly      & 1,884 & 0.48     & 11.17        & 1.87 ($n$=133) & 0       \\
        Scaffolded assembly & 1,065 & 0.86     & 95.24        & 48.92 ($n$=8)  & 2,536,928 \\ \bottomrule
        \end{tabular}
        \caption{Comparison of global statistics between the draft and scaffolded assembly. Computed using the program Assembly-stats.~\cite{Assemblystats2023}}
        \label{tab:global}
\end{minipage}
\end{table}

According to cytometric estimations, the size of the Thymus vulgaris haploid genome is 754.60 Mb (C-value = 0.77pg).~\cite{marieCytometricExercisePlant1993,PlantDNACvalues} The size of our 13 pseudo-chromosomes is 695.63 Mb. In addition, if we consider the sum of the 48 super contigs placed by homology with unplaced T. quinquecostatus contigs, the number is closer to the estimation:  706.41Mb. However, the size of the whole assembly exceeds the estimate; 911.87Mb.\\

\autoref{fig:gaps} shows the distribution of inferred gap sizes. It should be noted that most of the gaps have a length of 100, which by convention, indicates that we do not have enough information to establish the size of the gap.~\cite{AGPSpecificationV2} \\

\begin{figure}
    \centering
    \def\svgwidth{\textwidth}
    \input{gfx/gap_vs_chrom.pdf_tex}
    \caption{The inferred gap sizes plotted for each pseudo-chromosome, according to \eqref{eq:infergapsize}. The size of each pseudochromosome is shown to complement the information.}    
    \label{fig:gaps}
\end{figure}   

\section{Mapping short Illumina reads to scaffolded assembly}

\autoref{fig:error_rate} shows the result of mapping the Illumina reads of the five individuals to the scaffolded assembly. The two \textit{S. montana} individuals are grouped in the plot, showing similar mapping patterns to the scaffolded assembly of \textit{T. vulgaris}. They have a lower percentage of properly mapped sequences than the rest of \textit{T. vulgaris} specimens. The use of hybridization two, which enriches DNA in regions present in \textit{T. vulgaris}, increases the percentage of mapped reads and the number of mismatches in both \textit{S. montana} invidiuals. Although the increased number of mapped reads also affects \textit{T. vulgaris} individuals, it is less pronounced. The use of hybridization two does not increase the number of mismatches in \textit{T. vulgaris} as it does in \textit{S. montana}. The differences in the percentage of mapped reads between T. vulgaris individuals are slight, but the individual Thym607 used to construct the assembly has the lowest number of mismatches. Although we achieved better mapping results by penalizing SNPs and Indels more than the default parameters, we used default parameters for a more appropriate comparison.



\begin{figure}
    \centering
    \def\svgwidth{\textwidth}
    \input{gfx/mapped_vs_error_rate.pdf_tex}
    \caption{Aligned Illumina reads of 5 different invidiuals (see \autoref{tab:illumina_samples}) under two different hybridization conditions to the scaffolded assembly.}    
    \label{fig:error_rate}
\end{figure}   

\section{Mapping loci of interest to scaffolded assembly}

We aligned 